src/sqlfluff/dialects/dialect_bigquery.py

```diff
@@ -1827,6 +1828,44 @@ class AlterTableStatementSegment(ansi.AlterTableStatementSegment):
     )
 
 
+class AlterSchemaStatementSegment(BaseSegment):
+    """A `ALTER SCHEMA` statement."""
+
+    type = "alter_schema_statement"
+
+    match_grammar: Matchable = Sequence(
+        "ALTER",
+        "SCHEMA",
+        Ref("IfNotExistsGrammar", optional=True),
+        Ref("TableReferenceSegment"),
+        Sequence(
+            "SET",
+            OneOf(
+                # https://cloud.google.com/bigquery/docs/reference/standard-sql/data-definition-language#alter_schema_collate_statement
+                Ref("DefaultCollateSegment"),
+                # https://cloud.google.com/bigquery/docs/reference/standard-sql/data-definition-language#alter_schema_set_options_statement
+                Ref("OptionsSegment"),
+            ),
+            optional=True,
+        ),
+        # https://cloud.google.com/bigquery/docs/reference/standard-sql/data-definition-language#alter_schema_add_replica_statement
+        Sequence(
+            "ADD",
+            "REPLICA",
+            Ref("BaseExpressionElementGrammar"),
+            Ref("OptionsSegment", optional=True),
+            optional=True,
+        ),
+        # https://cloud.google.com/bigquery/docs/reference/standard-sql/data-definition-language#alter_schema_drop_replica_statement
+        Sequence(
+            "DROP",
+            "REPLICA",
+            Ref("BaseExpressionElementGrammar"),
+            optional=True,
+        ),
```

Comment:
Would it make sense to put these in a `OneOf` instead of the chained optional `Sequence`? They appear to be mutually exclusive options as far as I see.

Reply:
Good call 👍